content/docs/reference-events.md

```diff
@@ -165,11 +165,82 @@ onFocus onBlur
 
 These focus events work on all elements in the React DOM, not just form elements.
 
+#### onFocus
+
+The `onFocus` prop is initiated when focus is on the element. For example, when the user clicks on a text input element, the function defined with the `onFocus` prop is called.
+
+```
+  function onFocusExample() {
+     return (
+      <input
+        onFocus={(e) => {
+          console.log('Focused on input');
+        }}
+        placeholder="onFocus is triggered when you click this input."
+      />
+     )
+  }
+```
+
+#### onBlur
+
+`onBlur` prop is initiated when focus has left on the element. For example, when the user clicks into and then clicks outside of on a text input element, the function defined with the `onBlur` prop is called.
+
+```
+  function onBlurExample() {
+     return (
+      <input
+        onBlur={(e) => {
+          console.log('Triggered because this input lost focus');
+        }}
+        placeholder="onBlur is triggered when you click this input and then you click outside of it."
+      />
+     )
+  }
+```
+
 Properties:
 
 ```javascript
 DOMEventTarget relatedTarget
 ```
+`relatedTarget` is useful for detecting _child_ focus events as well as bubbling vs encapsulating events.
+
+You can use the `currentTarget` and `relatedTarget` to differentiate if the focusing or blurring events originated from _outside_ of the parent element:
+
+```
+export default function App() {
+  return (
+    <div
+      tabIndex={1}
+      onFocus={(e) => {
+        console.log("focused on", e.target);
+        if (e.currentTarget === e.target) {
+          console.log("focused on (self)");
+        }
+        if (!e.currentTarget.contains(e.relatedTarget)) {
+          console.log(
+            "Now focused on parent <div>, not triggered when swapping between children."
+          );
+        }
+      }}
+      onBlur={(e) => {
+        console.log("left focus of", e.target);
+        if (e.currentTarget === e.target) {
```

Comment:
I think this condition needs to be changed, because `e.currentTarget === e.target` will never be true for `onBlur` events as it is triggered when _self_ loses its focus or when the user leaves a specific field. Thus, this won't be true in any case.  

We can update it to this: 
```suggestion
        if (e.currentTarget !== e.target) {
```
> This will be triggered when we not focus on _self_ element anymore.

Reply:
I'm not sure what you mean but this condition is definitely true when you're blurring the node itself. https://codesandbox.io/s/strange-albattani-7tqr7?file=/src/App.js The proposed edit is wrong and contradicts what the code is supposed to do.